moveit_core/utils/src/message_checks.cpp

```diff
@@ -71,7 +71,7 @@ bool isEmpty(const moveit_msgs::msg::Constraints& constr)
 
 bool isEmpty(const geometry_msgs::msg::Quaternion& msg)
 {
-  return msg.x == 0.0 && msg.y == 0.0 && msg.z == 0.0 && msg.w == 0.0;
+  return msg.x == 0.0 && msg.y == 0.0 && msg.z == 0.0 && msg.w == 1.0;
```

Comment:
I think this should cover both cases. Whether `w` is initialized or not initialized.

```suggestion
  return (msg.x == 0.0 && msg.y == 0.0 && msg.z == 0.0 && msg.w == 1.0) || (msg.x == 0.0 && msg.y == 0.0 && msg.z == 0.0 && msg.w == 0.0);
```

Reply:
I don't think we should handle the other case since it no longer exist in ROS2